### PR TITLE
Introducing paparazzi-annotations module

### PR DESCRIPTION
**Pull request by nak5ive**
To be used in a follow-up PR, where we introduce a new KSP plugin.

### CODE REVIEW

File: paparazzi-annotations/build.gradle [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apply plugin: 'org.jetbrains.kotlin.jvm'
+apply plugin: 'com.vanniktech.maven.publish'
+
+dependencies {}
```
Comment (jrodbx) [inline paparazzi-annotations/build.gradle:4]: empty block can probably be removed?

File: paparazzi-annotations/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=paparazzi-annotations
+POM_NAME=Paparazzi Annotations API
```
Comment (jrodbx) [inline paparazzi-annotations/gradle.properties:2]:
```suggestion
POM_NAME=Paparazzi Annotations
```

**Review by jrodbx — APPROVED** [review]

File: paparazzi-annotations/src/main/java/app/cash/paparazzi/annotation/api/Paparazzi.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package app.cash.paparazzi.annotation.api
+
+@Target(AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.BINARY)
+annotation class Paparazzi
```
Comment (TWiStErRob) [inline paparazzi-annotations/src/main/java/app/cash/paparazzi/annotation/api/Paparazzi.kt:5]: Are you sure about the name? It seems a bit too generic to me, Potentially lending itself to being a god-annotation with many disparate parameters in the future.

Comment (jrodbx) [reply paparazzi-annotations/src/main/java/app/cash/paparazzi/annotation/api/Paparazzi.kt:5]: yep, we don't expect to use this outside of `@Preview` nor to add any parameters.  also, fine to break in the future if we're wrong.

**Pull request merged**